hooks/format-hook

```diff
@@ -0,0 +1,21 @@
+#!/bin/sh  
+  
+if git rev-parse --verify HEAD >/dev/null 2>&1  
+then  
+against=HEAD  
+else  
+# Initial commit: diff against an empty tree object  
+against=$(git hash-object -t tree /dev/null)
+fi  
+  
+# Test clang-format  
+clangformatout=$(git clang-format --diff --staged -q)  
```

Comment:
i just tried running this command on latest `main`, i.e. 32ba057, but it does not return anything :thinking: 
i'm not sure the `if` below will trigger :confused:

Reply:
This `if` does trigger, you can test the hook by cloning the PR/installing it, then break the indentation for example of some line and push it. It will report that the code is not formatted. No clue why it returns nothing tho

Reply:
the format was good before this PR?

Reply:
Tried it on main, it does return nothing if staged file are formatted, and stuff if not (ie needs formatting), so it definetly works

Reply:
mm ok, not a huge fan of just checking the staged files tbh